src/js/models/markup.js

```diff
@@ -22,8 +27,26 @@ export default class Markup {
     }
   }
 
+  // Use `create` to make a new markup so that we can use singletons in the
+  // markupMap
+  static create(tagName, attributes=[]) {
+    tagName = normalizeTagName(tagName);
+
+    if (attributes.length === 0) {
+      if (!markupMap[tagName]) {
+        markupMap[tagName] = new Markup(tagName);
```

Comment:
Oh right, this whole function should perhaps not exist (builder should be doing this instead).